frontend/src/app/offers/offers.component.ts

```diff
@@ -19,8 +20,10 @@ export class OffersComponent implements OnInit {
     .subscribe(
       offers => {
         this.offers = offers;
+        for (let offer of offers) {
+              offer.viewUrl = environment.djangoRoot + '/offers/'+ offer.slug + '/'+ offer.id;
```

Comment:
use template strings (https://developer.mozilla.org/pl/docs/Web/JavaScript/Referencje/template_strings) instead of old string concatination.

Reply:
Done.